oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -3039,3 +3039,255 @@ def sort(self, ascending=True):
+        >>> sa.rolling_sum(-2,0)
+        dtype: int 
+        Rows: 5
+        [None, None, 6, 9, 12]
+
+        Pandas equivalent:
+        >>> pandas.rolling_sum(series, 3)
+        0   NaN
+        1   NaN
+        2     2
+        3     3
+        4     4
+        dtype: float64
+
+        Same rolling sum operation, but 2 minimum observations:
+        >>> sa.rolling_sum(-2,0,min_observations=2)
+        dtype: int 
+        Rows: 5
+        [None, 3, 6, 9, 12]
+
+        Pandas equivalent:
+        >>> pandas.rolling_sum(series, 3, min_periods=2)
+        0    NaN
+        1      3 
+        2      6
+        3      9
+        4     12
+        dtype: float64
+
+        A rolling sum with a size of 3, centered around the current:
+        >>> sa.rolling_sum(-1,1)
+        dtype: int 
+        Rows: 5
+        [None, 6, 9, 12, None]
+
+        Pandas equivalent:
+        >>> pandas.rolling_sum(series, 3, center=True)
+        0   NaN
+        1     6
+        2     9
+        3    12
+        4   NaN
+        dtype: float64
+
+        A rolling sum with a window including the current and the 2 entries
+        following:
+        >>> sa.rolling_sum(0,2)
+        dtype: int
+        Rows: 5
+        [6, 9, 12, None, None]
+
+        A rolling sum with a window including the previous 2 entries NOT
+        including the current:
+        >>> sa.rolling_sum(-2,-1)
+        dtype: int 
+        Rows: 5
+        [None, None, 3, 5, 7]
+        """
+        if min_observations is None:
+            min_observations = (1 << 64) - 1
+        if min_observations < 0:
+            raise ValueError("min_observations must be a positive integer")
+        agg_op = None
+        if self.dtype() is array.array:
+            agg_op = '__builtin__vector__sum__'
+        else:
+            agg_op = '__builtin__sum__'
+        return SArray(_proxy=self.__proxy__.rolling_apply(agg_op, window_start, window_end, min_observations))
+
+    def rolling_max(self, window_start, window_end, min_observations=None):
+        if min_observations is None:
+            min_observations = (1 << 64) - 1
+        if min_observations < 0:
+            raise ValueError("min_observations must be a positive integer")
+        agg_op = '__builtin__max__'
+        return SArray(_proxy=self.__proxy__.rolling_apply(agg_op, window_start, window_end, min_observations))
+
+    def rolling_min(self, window_start, window_end, min_observations=None):
+        if min_observations is None:
+            min_observations = (1 << 64) - 1
+        if min_observations < 0:
+            raise ValueError("min_observations must be a positive integer")
+        agg_op = '__builtin__min__'
+        return SArray(_proxy=self.__proxy__.rolling_apply(agg_op, window_start, window_end, min_observations))
+
+    def rolling_var(self, window_start, window_end, min_observations=None):
+        if min_observations is None:
+            min_observations = (1 << 64) - 1
+        if min_observations < 0:
```

Comment:
Refactor this into a utility function. No need to copy paste this block of code so much :)